### DISCUSSION TITLE
Export API docs

### DISCUSSION THREAD

**Discussion by jaddison** [fastapi/fastapi#7853] [Questions] [answered]
Is it possible to have a method to extract or export the auto generated docs to a static set of pages? (This may already be possible, and I've missed it)

Thinking of separate docs subdomain and separating API and docs HTTP request traffic, etc.

**Comment by petervanderdoes** [comment, marked as answer]
It's a matter of copying the openapi.json and setting up which ever Doc System you wanna use, Swagger, Redoc. They both are just static one page websites.

**Comment by jaddison** [comment]
Is it that easy? That's pretty cool. Thanks - closing this issue.

**Comment by dmontagu** [comment]
@jaddison This section of the docs may also be of interest to you: https://fastapi.tiangolo.com/tutorial/extending-openapi/#self-hosting-javascript-and-css-for-docs

**Comment by pawamoy** [comment]
Page was moved to https://fastapi.tiangolo.com/advanced/extending-openapi/#self-hosting-javascript-and-css-for-docs

> It's a matter of copying the openapi.json and setting up which ever Doc System you wanna use, Swagger, Redoc. They both are just static one page websites.

Then, is there a way to generate and export the openapi.json, without having to actually run/serve the API? I would like to generate the JSON file in a CI job, keep it as an artifact, and deploy it somewhere else in another job. If I have to run the API to get the JSON file, it means I have to run the API in a background process and wget/curl the /openapi.json URL in a foreground process? It feels a bit weird to use background processes in CI.

**Comment by nsidnev** [comment]
@pawamoy 
You can call [`FastAPI.openapi()`](https://github.com/tiangolo/fastapi/blob/55afb70b3717969565499f5dcaef54b1f0acc7da/fastapi/applications.py#L81) method that will return OpenAPI schema as python dictionary. Check [here](https://fastapi.tiangolo.com/advanced/extending-openapi/#the-normal-process) for more information.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :cake: :bowing_man: 

And thanks @jaddison for reporting back and closing the issue :+1:

**Comment by pawamoy** [comment]
I posted a Python script [here](https://github.com/Redocly/redoc/issues/726) to export the ReDoc docs into a standalone HTML file, if anyone is interested :wink:
